### PR TITLE
Fix ApiApprovals and remove ObjectApprover

### PR DESCRIPTION
**Pull request by SeanFeldman**
Changes needed to reboot work on the failing `support-1.5` branch (will be coming short after this one is merged).

- Fix ApiApprover to remove dependency on TC to have the same agent compile and run tests.
- Remove dependency on ObjectApprover as it cannot take a hint for path

### CODE REVIEW

**Comment by SeanFeldman** [comment]
> There appears to be changes related to Cache management in this PR. Was that intentional?

TL;DR: there are no changes to the cache management implementation, but the tests verifying it's working.

Original tests used [ObjectApproval](https://github.com/Particular/NServiceBus.Persistence.AzureStorage/pull/191/files#diff-1cefd18488066c94984e51461a82296eL19) package, which does not allow overriding of the path where the source is found. Which will lead to failing tests when TC Compile and Test steps are not on the same agent. This results in tests failing with false positive.

Rather than forcing TC to use the same build agent, I opted into fixing the support branch to ensure when we address this with `develop`, we don't have to go back and fix the `support` branches.
